lib/switchtec.c

```diff
@@ -398,6 +398,101 @@ static const char *lane_reversal_str(int link_up,
 	}
 }
 
+
+
+
+
+/**
+ * @brief Get the LTSSM log of a port on a switchtec device
+ * @param[in]  dev    Switchtec device handle
+ * @param[in]  port   Switchtec Port
+ * @param[out] log    A pointer to an array containing the log 
+ *
+ */
+
+int switchtec_ltssm_log(struct switchtec_dev *dev,
+		int port, int *log_count,
+		ltssm_log_data *ltssm_log_output_data) 
+{
+	int ret = 1;
+
+	struct {
+		uint8_t sub_cmd;
+		uint8_t port;
+		uint8_t freeze;
+		uint8_t unused;
+	} ltssm_freeze;
+
+	ltssm_freeze.sub_cmd = 14;
+	ltssm_freeze.port = port;
+	ltssm_freeze.freeze = 1; //Freeze
+
+	ret = switchtec_cmd(dev, MRPC_DIAG_PORT_LTSSM_LOG, &ltssm_freeze, sizeof(ltssm_freeze), NULL, 0);
+
+	struct {
+		uint8_t sub_cmd;
+		uint8_t port;
+	}ltssm_status;
```

Comment:
Don't declare things in the middle of a function. Prefer it all at the top, or in this case it looks like this function should be split into multiple static helpers -- at least one per switchtec_cmd().